### PR TITLE
Revert to use the opster API's and onboard to OpenSearch universal release mechanism 

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description
Revert to use the opster API's and onboard to [OpenSearch universal release process ](https://github.com/opensearch-project/opensearch-build/blob/main/ONBOARDING.md#onboarding-to-universal--1-click-release-process).

### Issues Resolved
https://github.com/opensearch-project/opensearch-k8s-operator/issues/709 
https://github.com/opensearch-project/opensearch-k8s-operator/issues/674 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by prudhvigodithi** [comment]
Hey @swoehrl-mw can you please take a look and add your review? looks to me like the functional tests are failing because of the image not able to get pulled.
Thanks
Adding @bbarani @peterzhuamazon

File: jenkins/release.jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                [key: 'isTag', value: '$.ref_type'],
+            ],
+            tokenCredentialId: 'jenkins-opensearch-operator-generic-webhook-token',
+            causeString: 'A commit was pushed on opensearch-project/opensearch-k8s-operator repository main branch causing this workflow to run',
+            printContributedVariables: false,
+            printPostContent: false,
+        )
+    }
+    environment {
+        REPO_URL="https://github.com/prudhvigodithi/opensearch-k8s-operator"
```
Comment (swoehrl-mw) [inline jenkins/release.jenkinsfile:48]: Wrong url

**Review by swoehrl-mw — COMMENTED** [review]
@prudhvigodithi I suspect it's unable to pull the busybox image (for the init containers) as it does not yet exist.

File: jenkins/release.jenkinsfile (hunk 1 above)
Comment (prudhvigodithi) [reply jenkins/release.jenkinsfile]: Thanks I have updated the URL, I was testing on my fork.

**Comment by prudhvigodithi** [comment]
Next steps are:

- Onboard a jenkins job in https://build.ci.opensearch.org/ that uses the `jenkins/release.jenkinsfile`.
- Cut a tag with `2.5.0` which should trigger the above created job to build and push the docker images to ECR and dockerhub.
- The existing `release.yaml` will do a helm and github release.

@swoehrl-mw @bbarani @peterzhuamazon

**Review by swoehrl-mw — APPROVED** [review]

**Review by peterzhuamazon — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
Thanks @swoehrl-mw and @peterzhuamazon, one more change, I dont think we need to build, push and maintain the busybox image separately , I have modified the logic to use the existing official busybox https://hub.docker.com/_/busybox, any concerns ?

**Comment by prudhvigodithi** [comment]
Yup all the tests pass now, merging the PR.

**Comment by swoehrl-mw** [comment]
> I have modified the logic to use the existing official busybox https://hub.docker.com/_/busybox, any concerns ?

I believe it was only mirrored so that people would not potentially run into the dockerhub rate limits. But for most users this should not be a concern.

**Comment by prudhvigodithi** [comment]
Thanks I can change to ECR busybox https://gallery.ecr.aws/docker/library/busybox, as ECR has no rate limits for now.

**Pull request merged**